src/components/Story/Story.js

```diff
@@ -81,36 +98,57 @@ class Story extends React.Component {
     let followText = '';
 
     if (postState.userFollowed && !pendingFollow) {
```

Comment:
There seems to be a lot of logic here in the render function, it might be a good idea to create a helper function within this component called `this.getFollowText()` or something and have all the logic in there rather then in the render function.

Reply:
Although I agree with this I don't think that it's related to this PR and working on multiple unrelated pieces in the same PR shouldn't be encouraged. I don't think we should rewrite half of our code base while adding one feature. I think we should work on this in new PR.

Reply:
yeah good idea, this PR is already pretty big